runtime/runtime_impl_jetty9/src/main/java/com/google/apphosting/runtime/jetty9/JettyHttpHandler.java

```diff
@@ -0,0 +1,290 @@
+        break;
+      default:
+        throw new IllegalStateException(request.getRequestType().toString());
+    }
+  }
+
+  private void dispatchServletRequest(
+      String target, JettyRequestAPIData request, JettyResponseAPIData response)
+      throws Throwable {
+    Request baseRequest = request.getBaseRequest();
+    HttpServletRequest httpServletRequest = request.getHttpServletRequest();
+    HttpServletResponse httpServletResponse = response.getHttpServletResponse();
+    baseRequest.setAttribute(AppEngineConstants.APP_VERSION_KEY_REQUEST_ATTR, appVersionKey);
+
+    // Environment is set in a request attribute which is set/unset for async threads by
+    // a ContextScopeListener created inside the AppVersionHandlerFactory.
+    super.handle(target, baseRequest, httpServletRequest, httpServletResponse);
+  }
+
+  private void dispatchBackgroundRequest(JettyRequestAPIData request, JettyResponseAPIData response)
+      throws InterruptedException, TimeoutException {
+    String requestId = getBackgroundRequestId(request);
+    // The interface of coordinator.waitForUserRunnable() requires us to provide the app code with a
+    // working thread *in the same exchange* where we get the runnable the user wants to run in the
+    // thread. This prevents us from actually directly feeding that runnable to the thread. To work
+    // around this conundrum, we create an EagerRunner, which lets us start running the thread
+    // without knowing yet what we want to run.
+
+    // Create an ordinary request thread as a child of this background thread.
+    EagerRunner eagerRunner = new EagerRunner();
+    Thread thread = ThreadManager.createThreadForCurrentRequest(eagerRunner);
+
+    // Give this thread to the app code and get its desired runnable in response:
+    Runnable runnable =
+        coordinator.waitForUserRunnable(
+            requestId, thread, WAIT_FOR_USER_RUNNABLE_DEADLINE.toMillis());
+
+    // Finally, hand that runnable to the thread so it can actually start working.
+    // This will block until Thread.start() is called by the app code. This is by design: we must
+    // not exit this request handler until the thread has started *and* completed, otherwise the
+    // serving infrastructure will cancel our ability to make API calls. We're effectively "holding
+    // open the door" on the spawned thread's ability to make App Engine API calls.
+    // Now set the context class loader to the UserClassLoader for the application
+    // and pass control to the Runnable the user provided.
+    ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(appVersion.getClassLoader());
+    try {
+      eagerRunner.supplyRunnable(runnable);
+    } finally {
+      Thread.currentThread().setContextClassLoader(oldClassLoader);
+    }
+    // Wait for the thread to end:
+    thread.join();
+  }
+
+  private boolean handleException(
```

Comment:
remove bolean return, not used and failing internally for unused return value

Reply:
done